### DISCUSSION TITLE
I need to connect SOCKS Proxy server

### DISCUSSION THREAD

**Discussion by HardikMoon** [libssh2/libssh2#1167] [Ideas]
As per I have checked in libssh2 I'm unable to set any other socket rather than CFSocket at the time of session handshake. CFSocket itself it does not support SOCK5 TCP Proxying. 

Basically First I have to authenticate with proxy server using username and password then I need to forward ssh traffic through that proxy server to destination server.

StackoverFlow  :  https://stackoverflow.com/questions/76736701/how-can-achieve-dynamic-forwardingsock-in-ios-libssh2-not-supported-what-i

Kindly please help ASAP for the same and please suggest any alternatives.

**Comment by HardikMoon** [comment]
Any update on this? @vszakats

**Comment by vszakats** [comment]
I'm not an expert in this, also it seems nobody else had an answer so far.

As a community project, we are accepting PRs to help implementing this, if support for SOCKS proxies needs changes in libssh2 itself.

Also, have you tried libcurl? It supports both proxies and libssh2, so that might work.

Moving this thread to Discussions.

**Reply by bagder** [reply]
I'm with @vszakats here. libcurl can do SFTP over SOCKS with libssh2 just fine and it is not doing anything magic. libssh2 already has the necessary API.
